### ISSUE TITLE
For postgres, applying type information to a bind variable does not work.

### ISSUE THREAD

**Issue by besk-cerity** [pihvi/yesql#18]
I encountered this issue when using an `ANY` function as a replacement for an `IN` clause. I created a mocha test to illustrate the problem:
```
it('pg bind array using ANY clause with type cast', () => {
  const query = 'SELECT id::int FROM user WHERE id=any(:idList::int[]) and born > :year;'
  const data = {idList: [1, 2, 3], year: 2000}
  const expected = {
    text: 'SELECT id::int FROM user WHERE id=any($1::int[]) and born > $2;',
    values: [[1, 2, 3], 2000]
  };
  assert.deepEqual(yesql.pg(query)(data), expected)
})
```

This used to work when we were using v3.2.2.

**Comment by besk-cerity** [comment]
I had thought this was strictly an issue with casting to an array (I suspected the `[]` was throwing off the regex matching), but it will similarly fail any time you associate casting with a bind variable. This test will also fail:
```
it('pg bind variable with type cast', () => {
  const query = 'SELECT id::int FROM user WHERE id=:id::int and born > :year;'
  const data = {id: 1, year: 2000}
  const expected = {
    text: 'SELECT id::int FROM user WHERE id=$1::int and born > $2;',
    values: [1, 2000]
  };
  assert.deepEqual(yesql.pg(query)(data), expected)
})
```

**Comment by pihvi** [comment]
Thanks for the excellent bug report!
I'm working on this to bring back the same logic from version 3.

**Comment by pihvi** [comment]
Well finally I got this solved with a proper solution. So the old v3 logic is back and the new feature (quoted strings), that broke this in the first place, is now properly supported.

Can you verify this also solved your issue in version 4.1.2?

**Comment by besk-cerity** [comment]
Thanks. A quick test seems to indicate this works for us.
